### PR TITLE
Fix uniform real distribution 

### PR DESCRIPTION
**Pull request by haahh**
Related to https://github.com/boostorg/compute/issues/587. 

Now generated floating-point values are uniformly distributed on the interval `[a, b)` and no value is greater or equal to `b`.

I also added test that checks if results are in fact in `[a,b)` range (by using `range_test_engine` class that generates max random value or zeroes).

### CODE REVIEW

**Comment by haahh** [comment]
Travis-CI tests failed only because of `find_end` algorithm. It's fixed by https://github.com/boostorg/compute/pull/597.

File: test/test_uniform_real_distribution.cpp [hunk 1]
```diff
@@ -40,7 +43,60 @@ distribution.generate(vec.begin(), vec.end(), engine, queue);
...
+BOOST_AUTO_TEST_CASE(uniform_real_distribution_range)
+{
+    using boost::compute::lambda::_1;
+
+    boost::compute::vector<float> vec(32, context);
+
+    // initialize the range_test_engine
+    range_test_engine<boost::compute::uint_> engine(queue);
+
+    // setup the uniform distribution to produce floats between 1 and 100
```
Comment (kylelutz) [inline test/test_uniform_real_distribution.cpp]: Shouldn't this be between `0.9` and `1.0`?

Comment (haahh) [inline test/test_uniform_real_distribution.cpp]: You're right! I'll fix this in a minute.

Comment (haahh) [inline test/test_uniform_real_distribution.cpp]: Fixed.

**Pull request merged**